src/cloud-api-adaptor/podvm/files/usr/local/bin/setup-scratch-storage

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+[ -n "${DEBUG:-}" ] && set -o xtrace
+
+handle_error() {
+    local exit_code="${?}"
+    local line_number="${1:-}"
+    echo "error:"
+    echo "Failed at $line_number: ${BASH_COMMAND}"
+    exit "${exit_code}"
+}
+trap 'handle_error $LINENO' ERR
+
+die() {
+    local msg="$*"
+    echo >&2 "ERROR: $msg"
+    exit 1
+}
+
+# Check if daemon.json exists and parse configuration
+DAEMON_CONFIG_PATH="/run/peerpod/daemon.json"
+SCRATCH_ENABLED="false"
+ENCRYPTION_ENABLED="false"
+
+if [ -f "$DAEMON_CONFIG_PATH" ]; then
+    if command -v jq &>/dev/null; then
+        SCRATCH_ENABLED=$(jq -r '.["enable-scratch-disk"] // false' "$DAEMON_CONFIG_PATH")
+        ENCRYPTION_ENABLED=$(jq -r '.["enable-scratch-encryption"] // false' "$DAEMON_CONFIG_PATH")
+    else
+        # Fallback to grep if jq is not available
+        if grep -q '"enable-scratch-disk":.*true' "$DAEMON_CONFIG_PATH"; then
+            SCRATCH_ENABLED="true"
+        fi
+        if grep -q '"enable-scratch-encryption":.*true' "$DAEMON_CONFIG_PATH"; then
+            ENCRYPTION_ENABLED="true"
+        fi
+    fi
+fi
+
+echo "Scratch disk enabled: $SCRATCH_ENABLED"
+echo "Scratch encryption enabled: $ENCRYPTION_ENABLED"
+
+# Exit early if scratch disk is not enabled
+if [ "$SCRATCH_ENABLED" != "true" ]; then
+    echo "Scratch disk is disabled. Exiting."
+    exit 0
+fi
+
+# Check if the trusted_store device exists
+DISK_LABEL="trusted_store"
+DEV_PATH="/dev/disk/by-label/$DISK_LABEL"
+
+if [ ! -L "$DEV_PATH" ]; then
+    echo "Device for label $DISK_LABEL not found. Exiting."
+    exit 0
```

Comment:
Should this really exit with 0 when the label is not found?

Reply:
Good catch, forgot to disable it post testing..